python/tank/bootstrap/configuration.py

```diff
@@ -78,18 +78,21 @@ def get_tk_instance(self, sg_user):
         path = self._path.current_os
         core_path = get_core_python_path_for_config(path)
 
+        # Get the user before the core swapping.
+        from ..authentication import serialize_user
```

Comment:
Good point, I should serialize first. I will however have to pass down the unserialized user as well as the core I am swapping into might not understand the serialized payload. Image a serialized SAML user pushed to a core that doesn't understand SAML for example.